packages/server/src/service-worker.ts

```diff
@@ -151,6 +154,8 @@ async function broadcastOne(request: Request): Promise<Response> {
     broadcast.addEventListener('message', messageHandler);
   });
 
+  // Add URL pathname to message
+  message.pathname = url.pathname;
```

Comment:
Note adding the `url.pathname` to messages sent from ServiceWorker to `ServiceWorkerManager`. This is so `/stdin/kernel` and be distinguished from `/stdin/terminal` as otherwise this information does not get to the `ServiceWorkerManager`. The alternative is to not pass it and instead change the `registerStdinHandler` to take another function which is a callback to check if a particular stdin message should be handled by that handler by checking some specifics of the `data` passed. For example, the kernel stdin handler would check that `header.msg_type === 'input_request'`.